src/openms/source/KERNEL/ConsensusMap.cpp

```diff
@@ -730,6 +732,99 @@ OPENMS_THREAD_CRITICAL(oms_log)
     return true;
   }
 
+  void ConsensusMap::split(std::vector<FeatureMap>& fmaps, ConsensusMap::SplitMeta mode) const
+  {
+    // Prepare output vector
+    fmaps.clear();
+    Size numbr_exps = column_description_.size();
+    fmaps.resize(numbr_exps);
+
+    // Check for Isobaric Analyzer
+    bool iso_analyze = true;
+    auto is_not_elem = [](const OpenMS::DataProcessing& dp)
+    {
+      return (dp.getProcessingActions().count(DataProcessing::QUANTITATION) == 0);
+    };
+    auto cm_dp = (*this).getDataProcessing(); // get a copy to avoid calling .begin() and .end() on two different temporaries
+    if (all_of(cm_dp.begin(), cm_dp.end(), is_not_elem))
+    {
+      iso_analyze = false;
+    }
+
+    for (const auto& cf : *this)
+    {
+      // Create new Features from FeatureHandles
+      std::vector<BaseFeature> new_feats(numbr_exps);
+      for (const FeatureHandle& fh : cf.getFeatures())
+      {
+        BaseFeature feat(fh);
+        new_feats[fh.getMapIndex()] = feat;
+      }
+
+      // Add PeptideIdentifications to ...
+      for (const PeptideIdentification& pep_id : cf.getPeptideIdentifications())
+      {
+
+        if (iso_analyze)
+        {
+          // ... the first Feature.
+          new_feats[0].getPeptideIdentifications().push_back(pep_id);
+          continue;
+        }
+
+        // ... the corresponding Feature by map_index.
+        if (!pep_id.metaValueExists("map_index"))
+        {
+          throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
+              "File did not undergo isobaric analyzation, but no map index was found at PeptideIdentifications. Check Input!");
+        }
+        new_feats[pep_id.getMetaValue("map_index")].getPeptideIdentifications().push_back(pep_id);
+      }
+
+      // Handle MetaValues
+      switch(mode)
+      {
+        case SplitMeta::IGNORE : 
+          break;
+
+        case SplitMeta::COPY_ALL :
+          for (BaseFeature& base : new_feats)
+          {
+            base.MetaInfoInterface::operator=(cf);
+          }
+          break;
+
+        case SplitMeta::COPY_FIRST :
+          new_feats[0].MetaInfoInterface::operator=(cf);
+          break;
+      }
+
+      // Add new Features to corresponding FeatureMap.
+      for (Size i = 0; i < numbr_exps; ++i)
+      {
+        fmaps[i].emplace_back(new_feats[i]);
```

Comment:
emplace does not help here.
You could do `.push_back(std::move(new_feats[i]);` for some speed gain

Reply:
`.push_back(std::move(new_feats[i]));` does not work
i get the same compile error i got while just using `.push_back(new_feats[i]);`:
"no matching member function"

Reply:
ok, forget what I said. All ok.
To make it more performant one would need an R-value CTor for BaseFeature in Feature.h, i.e. `Feature(BaseFeature&& b)` and then ´.emplace_back(std::move(new_feats[i]);´